com.civicrm.hrjobroles/js/src/job-roles/controllers/hr-job-roles-controller.js

```diff
@@ -312,6 +310,40 @@ define([
                         }
                     }
                 }
+
+                if(form_id == 'end_date' && !$scope.edit_data[role_id].end_date){
+                    $scope.edit_data[role_id].end_date = null;
+                }
+
+                if ($scope.edit_data[role_id].job_contract_id
+                    && $scope.edit_data[role_id].start_date
+                    && typeof $scope.edit_data[role_id].end_date != 'undefined'
+                    && (form_id == 'start_date' || form_id == 'job_contract_id' || form_id == 'end_date')) {
+
+                    $scope.updateRolesWithContractData(role_id);
+                }
+            };
+
+            $scope.updateRolesWithContractData = function (role_id) {
+                var contract_id = $scope.edit_data[role_id].job_contract_id;
+
+                if ($scope.checkIfDatesAreCustom($scope.edit_data[role_id]['start_date'], $scope.edit_data[role_id]['end_date'])) {
+                    var contract = me.contractsData[contract_id];
+
+                    // search for revision containing these dates
+                    var revision = contract.revisions.filter(function (rev) {
```

Comment:
Since you are checking for `revision.length` (without then actually using `revision`) afterwards, you could use ES5 `some`
```js
var foo = contract.revisions.some(function (revision) {
   return //...
});

if (foo) {
    // ...
}